diff --git a/src-os/ptime_clock_stubs.c b/src-os/ptime_clock_stubs.c
index 79052cc..d657b5f 100644
--- a/src-os/ptime_clock_stubs.c
+++ b/src-os/ptime_clock_stubs.c
@@ -17,22 +17,9 @@
 
 /* Detect platforms and call in their includes */
 
-#if defined(__APPLE__) && defined(__MACH__)
-  #define OCAML_PTIME_DARWIN
-  #include <time.h>
-  #include <sys/time.h>
-
-#elif defined (__unix__) || defined(__unix)
  #include <unistd.h>
- #if defined(_POSIX_VERSION)
-   #define OCAML_PTIME_POSIX
-   #include <time.h>
- #endif
-
-#else
-  #warning OCaml Ptime_clock module: unsupported platform
-  #define OCAML_PTIME_UNSUPPORTED
-#endif
+ 
+ 
 
 
 /* Clock now */