diff --git a/pkg/pkg.ml b/pkg/pkg.ml
index fe02f0b..0d454ca 100755
--- a/pkg/pkg.ml
+++ b/pkg/pkg.ml
@@ -19,7 +19,5 @@ let () =
        Pkg.mllib ~cond:jsoo "src/logs_browser.mllib";
        Pkg.mllib ~cond:cmdliner "src/logs_cli.mllib";
        Pkg.mllib ~cond:lwt "src/logs_lwt.mllib";
-       Pkg.mllib ~cond:fmt ~api:[] "src/logs_top.mllib";
-       Pkg.lib "src/logs_top_init.ml";
        Pkg.doc "test/tool.ml";
        Pkg.doc "test/tags.ml"; ]
