diff --git a/src/tcpip_checksum/jbuild b/src/tcpip_checksum/jbuild
index f12101d..d8e34f6 100644
--- a/src/tcpip_checksum/jbuild
+++ b/src/tcpip_checksum/jbuild
@@ -11,6 +11,7 @@
  ((name tcpip_xen)
   (public_name tcpip.xen)
   (libraries (tcpip))
+  (no_dynlink)
   (modules (tcpip_xen))
   (c_names (checksum_stubs_xen))
   (c_flags (:include c_flags_xen.sexp))
@@ -18,8 +19,7 @@
 
 (rule
  ((targets (c_flags_xen.sexp))
-  (deps (../config/discover.exe))
-  (action (run ${<} -ocamlc ${OCAMLC}))))
+  (action (run ${exe:../config/discover.exe} -ocamlc ${OCAMLC}))))
 
 (rule
   ((targets (checksum_stubs_xen.c))
@@ -30,5 +30,6 @@
  ((name tcpip_unix)
   (public_name tcpip.unix)
   (modules (tcpip_unix))
+  (no_dynlink)
   (c_names (checksum_stubs))
   (wrapped false)))
